moveit_core/smoothing_plugins/include/moveit/smoothing_plugins/smoothing_base_class.h

```diff
@@ -0,0 +1,72 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2021, PickNik Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Andy Zelenak
+   Description: Defines a pluginlib interface for smoothing algorithms.
+ */
+
+#pragma once
+
+#include <moveit/robot_model/robot_model.h>
+
+namespace smoothing_plugins
+{
+class SmoothingBaseClass
+{
+public:
+  /**
+   * Initialize the smoothing algorithm
+   * @param node ROS node, typically used for parameter retrieval
+   * @param robot_model typically used to retrieve vel/accel/jerk limits
+   * @param num_joints number of actuated joints in the JointGroup Servo controls
+   * @return True if initialization was successful
+   */
+  virtual bool initialize(rclcpp::Node::SharedPtr node, moveit::core::RobotModelConstPtr robot_model,
+                          size_t num_joints) = 0;
+
+  /**
+   * Smooth an array of joint position deltas
+   * @param delta_theta array of incremental joint position commands
```

Comment:
```suggestion
   * @param position_vector array of incremental joint position commands
```

Reply:
I was wondering if people usually duplicate the documentation when they inherit from a base class. This is a good example where I had to make your change in 2 places. What do you think?

Reply:
I personally don't like having to dig my way up a interface chain to find documentation (granted, this is just 1 level). Maybe I could be using my IDE better but my preference would be to have the doc here as well.

Reply:
In particular I think some of the core ROS stuff is annoying to trace and up find the units being used (which is one reason why I am on the bandwagon of including the unit in the variable name)